internal/staticanalysis/linelengths/line_lengths.go

```diff
@@ -0,0 +1,67 @@
+package linelengths
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"strings"
+)
+
+/*
+GetLineLengths counts the number of characters on each line of a file or string,
+returning a map of length to the number of lines in the file with that length.
+
+Lines are defined to be separated by newline ('\n') characters. If the newline
+character is preceded by a carriage return ('\r'), this will also be treated as
+part of the separator.
+
+If filePath is not empty, the function attempts to count the lines of the file
+at that path, otherwise lines in sourceString are counted.
+*/
+func GetLineLengths(filePath string, sourceString string) (map[int]int, error) {
+	var reader *bufio.Reader
+	if len(filePath) > 0 {
+		file, err := os.Open(filePath)
+		if err != nil {
+			return nil, err
+		}
+		defer file.Close()
+
+		reader = bufio.NewReader(file)
+	} else {
+		reader = bufio.NewReader(strings.NewReader(sourceString))
+	}
+
+	lengths := map[int]int{}
+	for {
+		line, readErr := reader.ReadString('\n')
```

Comment:
Can we use https://pkg.go.dev/bufio#example-Scanner-Lines  instead here? That can make things a lot simpler. 

Or if there's a reason to not use it, document it in a comment.

Reply:
though I wonder if this limitation also applies to `ReadString`...

Reply:
`ReadString` handles large buffers better out of the box - I thought it would be simpler to have code dealing with newline characters than dealing with buffers

Reply:
Will add a comment in another PR

Reply:
Sure I'll add a comment.

If you read the code, `ReadString` uses a function called `collectFragment` which is able to aggregate multiple full buffers. Whereas `bufio.Scanner` by default uses a `MaxScanTokenSize = 64 * 1024`